db/migrate/20151208150956_fix_host_storage_replication_on_upgrade.rb

```diff
@@ -0,0 +1,49 @@
+class FixHostStorageReplicationOnUpgrade < ActiveRecord::Migration
+  include MigrationHelper
+  include MigrationHelper::SharedStubs
+
+  class MiqRegion < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class HostsStorage < ActiveRecord::Base
+    self.table_name = "host_storages"
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    if on_replication_target?
+      run_for_replication_target
+    else
+      run_for_replication_source
+    end
+  end
+
+  def on_replication_target?
+    MiqRegion.select(:region).distinct.count > 1
+  end
+
+  def run_for_replication_target
+    HostsStorage.delete_all
```

Comment:
Same as above, I don't remember if we need to remove a scope to see all regions hosts_storages and remove them